packages/expo-dev-launcher/android/src/main/java/com/facebook/react/devsupport/DevLauncherSettings.kt

```diff
@@ -2,18 +2,21 @@ package com.facebook.react.devsupport
 
 import android.content.Context
 import com.facebook.react.packagerconnection.PackagerConnectionSettings
-import expo.modules.devlauncher.react.DevLauncherPackagerConnectionSettings
 
 internal class DevLauncherSettings(
   context: Context,
   debugServerHost: String
 ) : DevMenuSettingsBase(context) {
-  private var connectionSettings = DevLauncherPackagerConnectionSettings(context, debugServerHost)
+  private var connectionSettings = PackagerConnectionSettings(context)
   override val packagerConnectionSettings: PackagerConnectionSettings
     get() = connectionSettings
 
   // Implemented here so `this` is not leaked
   init {
+    // We can't extend PackagerConnectionSettings anymore, because now it's final class
+    // So we need to update the debugServerHost on init
+    packagerConnectionSettings.debugServerHost = debugServerHost
```

Comment:
doing this will overwrite user's sharedpreference: https://github.com/facebook/react-native/blob/b69a92e2c95990fcfa95a6ab7b7260ac492f942c/packages/react-native/ReactAndroid/src/main/java/com/facebook/react/packagerconnection/PackagerConnectionSettings.kt#L41
especially when developing dev-menu or dev-launcher with local dev-server, that would mess up the dev server host all together from a single sharedpreference

Reply:
ohh I see, should we just use reflection then?

Reply:
i don't know tbh. the only solution in my head is to send a pr to upstream. maybe extract the `PackagerConnectionSettings` as an interface